consensus/src/checkpoint.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::{Chain, ProtocolStateChain};
+use mina_crypto::base58::Base58Encodable;
+use mina_crypto::hash::{EpochSeed, StateHash};
+use mina_rs_base::protocol_state::ProtocolState;
+
+pub const SLOTS_PER_EPOCH: u32 = 7140;
+
+pub fn init_checkpoints(genesis: &mut ProtocolState) {
```

Comment:
why not impl init_checkpoints  and is_short_range  for ProtocolState and ProtocolStateChain respectively

Reply:
They're not pure functions as params are mutable